File: frontend/src/types/factura.ts
export type MetodoPago =
  | "efectivo"
  | "tarjeta"
  | "transferencia"
  | "nequi"
  | "daviplata";

export const METODOS_PAGO: MetodoPago[] = [
  "efectivo",
  "tarjeta",
  "transferencia",
  "nequi",
  "daviplata",
];

export interface Factura {
  id: number;
  id_pedido: number;
  numero_factura: string;
  fecha_emision: string;
  subtotal: string;
  impuestos: string;
  total: string;
  metodo_pago: MetodoPago;
  created_at: string;
}

export interface FacturaCreate {
  metodo_pago: MetodoPago;
}

export interface FacturasPaginadas {
  items: Factura[];
  total: number;
}

File: backend/app/schemas/factura.py
from datetime import datetime
from decimal import Decimal

from pydantic import BaseModel, ConfigDict

from app.models.enums import MetodoPago


class FacturaCreate(BaseModel):
    metodo_pago: MetodoPago


class FacturaOut(BaseModel):
    id: int
    id_pedido: int
    numero_factura: str
    fecha_emision: datetime
    subtotal: Decimal
    impuestos: Decimal
    total: Decimal
    metodo_pago: MetodoPago
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)


class FacturaListOut(BaseModel):
    items: list[FacturaOut]
    total: int
